benchmarks/cpp_micro_benchmarks.py

```diff
@@ -114,47 +88,23 @@ class RecordCppMicroBenchmarks(_benchmark.Benchmark):
     description = "Run the Arrow C++ micro benchmarks."
     iterations = 1
     flags = {"language": "C++"}
+    adapter = None
 
-    def run(self, **kwargs):
-        with tempfile.NamedTemporaryFile(delete=False) as result_file:
-            run_command = get_run_command(result_file.name, kwargs)
-            self.execute_command(run_command)
-            results = json.load(result_file)
-            for suite in results["suites"]:
-                self.conbench.mark_new_batch()
-                for result in suite["benchmarks"]:
-                    yield self._record_result(suite, result, kwargs)
-
-    def _record_result(self, suite, result, options):
-        info, context = {}, {"benchmark_language": "C++"}
-        tags = _parse_benchmark_name(result["name"])
-        name = tags.pop("name")
-        tags["suite"] = suite["name"]
-        tags["source"] = self.name
-
-        values = self._get_values(result)
-
-        return self.record(
-            values,
-            tags,
-            info,
-            context,
-            options=options,
-            output=result,
-            name=name,
+    def __init__(self):
+        tags, info, context = self._get_tags_info_context(case=None, extra_tags={})
```

Comment:
Same thing as below, a comment about where | why we are depending on this here might be nice

Reply:
Done. This appends a bunch of metadata about arrow like build flags, all extracted from pyarrow. I don't love that for a C++ benchmark, but a. it's consistent, and b. I'm not sure how else to dig it out.

Reply:
_nods_ yeah it's a bit odd, but in the end probably the easiest. We run into a similar issue with velox, where there's not (yet) an easy thing to poll for these kinds of things. But they are so project-specific it's not something we can really abstract.